solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -16,18 +17,38 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
     }
 
     @Override
-    public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
-    }
+    public void initialize() {}
 
     @Override
     public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + filename);
+            ObjectOutputStream out = new ObjectOutputStream(fos)){
+            out.writeObject(gameState); }
+        catch (FileNotFoundException ex) {
+            throw new SaveGameException("File not found", ex, SaveGameException.Type.USER, gameState);
+        }
+        catch (IOException ex) {
+            throw new SaveGameException("IO error", ex, SaveGameException.Type.IO, gameState);
+        }
     }
 
     @Override
     public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+        Savable load = null;
+        try (FileInputStream fis = new FileInputStream(filesDirectory + filename);
+             ObjectInputStream in = new ObjectInputStream(fis)) {
+             load = (Savable) in.readObject();
+        }
+        catch (FileNotFoundException ex) {
+            throw new SaveGameException("File not found", ex, SaveGameException.Type.USER, load);
+        }
```

Comment:
&#39;}&#39; at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a766953743>*